src/node.cc

```diff
@@ -2257,6 +2258,19 @@ inline napi_addon_register_func GetNapiInitializerCallback(DLib* dlib) {
       reinterpret_cast<napi_addon_register_func>(dlib->GetSymbolAddress(name));
 }
 
+// Initialize the addon, and close the DLib if this is a second-or-later
+// initialization call.
+#define CALL_ADDON_INIT(call, entry_point, dlib)              \
+  do {                                                        \
+    call;                                                     \
+    void* key_value = reinterpret_cast<void*>((entry_point)); \
+    if (addon_entry_points.count((key_value)) == 1) {         \
+      (dlib).Close();                                         \
+    } else {                                                  \
+      addon_entry_points.insert((key_value));                 \
```

Comment:
Can be DRY'd to this:
```c+
if (!addon_entry_points.insert(key_value).second) dlib.Close();
```
Apropos the name `key_value`, that to me suggests it's a `std::pair` when it's not.